third-party/cmake/BuildLuajit.cmake

```diff
@@ -51,30 +52,58 @@ if(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
 else()
   set(AMD64_ABI "")
 endif()
-set(INSTALLCMD_UNIX ${MAKE_PRG} CFLAGS=-fPIC
-                                CFLAGS+=-DLUA_USE_APICHECK
-                                CFLAGS+=-funwind-tables
-                                ${NO_STACK_CHECK}
-                                ${AMD64_ABI}
-                                CCDEBUG+=-g
-                                Q=
-                                install)
+set(BUILDCMD_UNIX ${MAKE_PRG} CFLAGS=-fPIC
+                              CFLAGS+=-DLUA_USE_APICHECK
+                              CFLAGS+=-funwind-tables
+                              ${NO_STACK_CHECK}
+                              ${AMD64_ABI}
+                              CCDEBUG+=-g
+                              Q=)
 
-if(UNIX)
-  if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    if(CMAKE_OSX_DEPLOYMENT_TARGET)
-      set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}")
-    else()
-      # Use the same target as our nightly builds
-      set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=10.11")
-    endif()
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  if(CMAKE_OSX_DEPLOYMENT_TARGET)
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}")
   else()
-    set(DEPLOYMENT_TARGET "")
+    # Use the same target as our nightly builds
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=10.11")
   endif()
+else()
+  set(DEPLOYMENT_TARGET "")
+endif()
 
-  BuildLuaJit(INSTALL_COMMAND ${INSTALLCMD_UNIX}
+if(UNIX AND NOT CMAKE_OSX_ARCHITECTURES)
+  BuildLuaJit(INSTALL_COMMAND ${BUILDCMD_UNIX}
     CC=${DEPS_C_COMPILER} PREFIX=${DEPS_INSTALL_DIR}
-    ${DEPLOYMENT_TARGET})
+    ${DEPLOYMENT_TARGET} install)
+
+elseif(CMAKE_OSX_ARCHITECTURES AND APPLE)
```

Comment:
should probably be an error of CMAKE_OSX_ARCHITECTURES is defined and APPLE is not?

Reply:
The CMake documentation states that `CMAKE_OSX_ARCHITECTURES` is ignored on inapplicable platforms, and the intention of this check is to follow that behaviour.

That said, I don't think this works as intended, because I think you end up not building LuaJIT if you define `CMAKE_OSX_ARCHITECTURES` on Linux. Will need to fix that.